libs/langchain-google-common/src/utils/anthropic.ts

```diff
@@ -0,0 +1,723 @@
+import {
+  ChatGeneration,
+  ChatGenerationChunk,
+  ChatResult,
+} from "@langchain/core/outputs";
+import {
+  BaseMessage,
+  BaseMessageChunk,
+  AIMessageChunk,
+  MessageContentComplex,
+  MessageContentText,
+  MessageContent,
+  MessageContentImageUrl,
+  AIMessageFields,
+  AIMessageChunkFields,
+} from "@langchain/core/messages";
+import {
+  ToolCall,
+  ToolCallChunk,
+  ToolMessage,
+} from "@langchain/core/messages/tool";
+import {
+  AnthropicAPIConfig,
+  AnthropicContent,
+  AnthropicContentText,
+  AnthropicContentToolUse,
+  AnthropicMessage,
+  AnthropicMessageContent,
+  AnthropicMessageContentImage,
+  AnthropicMessageContentText,
+  AnthropicMessageContentToolResult,
+  AnthropicMessageContentToolResultContent,
+  AnthropicRequest,
+  AnthropicRequestSettings,
+  AnthropicResponseData,
+  AnthropicResponseMessage,
+  AnthropicStreamContentBlockDeltaEvent,
+  AnthropicStreamContentBlockStartEvent,
+  AnthropicStreamInputJsonDelta,
+  AnthropicStreamMessageDeltaEvent,
+  AnthropicStreamMessageStartEvent,
+  AnthropicStreamTextDelta,
+  AnthropicTool,
+  AnthropicToolChoice,
+  GeminiTool,
+  GoogleAIAPI,
+  GoogleAIModelParams,
+  GoogleAIModelRequestParams,
+  GoogleAIToolType,
+  GoogleLLMResponse,
+} from "../types.js";
+
+export function getAnthropicAPI(config?: AnthropicAPIConfig): GoogleAIAPI {
+  // function notImplemented(): never {
+  //   throw new Error("Not implemented");
+  // }
+
+  function partToString(part: AnthropicContent): string {
+    return "text" in part ? part.text : "";
+  }
+
+  function messageToString(message: AnthropicResponseMessage): string {
+    const content: AnthropicContent[] = message?.content ?? [];
+    const ret = content.reduce((acc, part) => {
+      const str = partToString(part);
+      return acc + str;
+    }, "");
+    return ret;
+  }
+
+  function responseToString(response: GoogleLLMResponse): string {
+    const data = response.data as AnthropicResponseData;
+    switch (data?.type) {
+      case "message":
+        return messageToString(data as AnthropicResponseMessage);
+      default:
+        throw Error(`Unknown type: ${data?.type}`);
+    }
+  }
+
+  /**
+   * Normalize the AIMessageChunk.
+   * If the fields are just a string - use that as content.
+   * If the content is an array of just text fields, turn them into a string.
+   * @param fields
+   */
+  function newAIMessageChunk(fields: string | AIMessageFields): AIMessageChunk {
```

Comment:
Think it's worth reusing some logic from `@langchain/anthropic`?

https://github.com/langchain-ai/langchainjs/tree/main/libs/langchain-anthropic/src/utils

Could even make it a dep

Reply:
I considered it, and even tried for a bit, but ultimately decided not to for a few reasons:
* It adds an unnecessary dependency in the case where people don't intend to use Claude on GCP. Including pulling in the Anthropic library.
* Some of the types aren't exactly compatible (I don't remember which ones now - I just remember starting down this road and running into issues).

The code you highlighted here has more to do with how my logic was generating the content objects and how there are expectations in parts of the base code that `AIMessageChunk.content` should be a string. So this normalizes it in those cases. (Which are almost all the cases for AI Messages historically.)

Reply:
(To be clear, however, I'm open minded on this. I don't like duplicating code.)